### DISCUSSION TITLE
How do you disable the pure SSH protocol?

### DISCUSSION THREAD

**Discussion by antdking** [git-lfs/git-lfs#5133] [Q&A] [answered]
Currently tracing why `git push` is being slow for me, and it appears to be from git-lfs trying to connect to Github via SSH.

```
13:43:30.001964 trace git-lfs: attempting pure SSH protocol connection
13:43:30.001987 trace git-lfs: run_command: ssh git@github.com git-lfs-transfer datapane/datapane-hosted.git upload
13:43:30.002043 trace git-lfs: exec: ssh 'git@github.com' 'git-lfs-transfer datapane/datapane-hosted.git upload'
13:43:46.437537 trace git-lfs: pure SSH protocol connection failed: Unable to negotiate version with remote side (unable to read capabilities): EOF
```

This check runs each time, and takes anywhere between 5 and 20 seconds (16s in this case).

How do we force the fallback behaviour by default? (ie: use `git-lfs-authenticate`)

**Comment by bk2204** [comment, marked as answer]
Hey,

There's no way to disable it.  It's supposed to happen once at the beginning of the connection and fail quickly if it's not enabled.  It should definitely not take 5–20 seconds to connect in the normal situation, and if it's doing that, I suspect it's also going to do that for any sort of Git operation over SSH as well.  Do you also see that when you perform regular Git operations over SSH?

**Comment by antdking** [comment]
Git operations following seem fine, taking the expected ~1s for an SSH connection on a push.

It's typical that now I'm digging into the problem, it has since vanished[^1]
This looks like an intermittent issue, as I'm back to ~1-2s for detecting the failure; though this has been ongoing for a couple of weeks now.
I'll get more traces if/when it comes back. There's not much that can be done otherwise.

It'd still be good to have a way to disable the pure transport when the remote (Github, in this case) doesn't support it. I can imagine this would help people on less-than-ideal connections, too.

[^1]: Only difference was running `git lfs uninstall; git lfs install`
